File: Allspice/Models/Favorite.cs

namespace Allspice.Models;

public class Favorite
{
    public int Id { get; set; }
    public string AccountId { get; set; }
    public int RecipeId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public class RecipeFavorite : Favorite
{
  public int FavoriteId { get; set; }
}

public class AccountFavorite : Account
{
  public int FavoriteId { get; set; }
}

File: Allspice.client/src/models/Favorite.js
export class Favorite {
  constructor(data) {
    this.id = data.id
    this.favoriteId = data.favoriteId
    this.accountId = data.accountId
    this.recipeId = data.recipeId
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
  }
}

